### PR TITLE
upgrade to OZ v5 while keeping old version for fork upgrade tests

### PR DESCRIPTION
**Pull request by jonwalch**

### CODE REVIEW

File: script/DeployScript.s.sol [hunk 1]
```diff
@@ -22,7 +22,7 @@ contract DeployScript is Script {
         vm.startBroadcast(deployer);
 
         // deploy proxy admin contract
-        proxyAdmin = new ProxyAdmin();
+        proxyAdmin = new ProxyAdmin(admin);
```
Comment (cridmann) [inline script/DeployScript.s.sol:25]: Do we want to keep the old version of the deploy script that uses the old version of OZ?

File: script/DeployUsccScript.s.sol [hunk 2]
```diff
@@ -1,8 +1,8 @@
 pragma solidity ^0.8.26;
 
 import "forge-std/Script.sol";
-import "openzeppelin-contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
-import "openzeppelin-contracts/proxy/transparent/ProxyAdmin.sol";
+import "openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
```
Comment (cridmann) [inline script/DeployUsccScript.s.sol:4]: same

File: script/DeployUstbUpgrade.s.sol [hunk 3]
```diff
@@ -1,7 +1,7 @@
 pragma solidity ^0.8.20;
 
 import "forge-std/Script.sol";
-import "openzeppelin-contracts/proxy/transparent/ProxyAdmin.sol";
+import "openzeppelin-contracts/contracts/proxy/transparent/ProxyAdmin.sol";
```
Comment (cridmann) [inline script/DeployUstbUpgrade.s.sol:4]: same

File: src/SuperstateToken.sol [hunk 4]
```diff
@@ -5,12 +5,15 @@ import {IERC20Upgradeable} from "openzeppelin-contracts-upgradeable/interfaces/I
...
 import {Ownable2StepUpgradeable} from "openzeppelin-contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
 import {PausableUpgradeable} from "openzeppelin-contracts-upgradeable/security/PausableUpgradeable.sol";
-import {ECDSA} from "openzeppelin-contracts/utils/cryptography/ECDSA.sol";
+import {ECDSA} from "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
 
 import {ISuperstateToken} from "src/interfaces/ISuperstateToken.sol";
 import {IERC7246} from "src/interfaces/IERC7246.sol";
 import {AllowList} from "src/AllowList.sol";
 
+import {SuperstateOracle} from "onchain-redemptions/src/oracle/SuperstateOracle.sol";
```
Comment (cridmann) [inline src/SuperstateToken.sol:14]: is this being used?

File: test/SuperstateTokenTestBase.t.sol [hunk 5]
```diff
@@ -14,6 +16,8 @@ import {AllowList} from "src/AllowList.sol";
 import "test/AllowListV2.sol";
 import "test/USTBV2.sol";
 
+import {console} from "forge-std/console.sol";
```
Comment (cridmann) [inline test/SuperstateTokenTestBase.t.sol:19]: remove

File: test/SuperstateTokenTestBase.t.sol [hunk 6]
```diff
@@ -41,24 +46,32 @@ contract SuperstateTokenTestBase is Test {
...
         eve = vm.addr(evePrivateKey);
 
         AllowList permsImplementation = new AllowList(address(this));
 
-        // deploy proxy admin contract
-        proxyAdmin = new ProxyAdmin();
-
         // deploy proxy contract and point it to implementation
-        permsProxy = new TransparentUpgradeableProxy(address(permsImplementation), address(proxyAdmin), "");
+        permsProxy = new TransparentUpgradeableProxy(address(permsImplementation), address(this), "");
```
Comment (cridmann) [inline test/SuperstateTokenTestBase.t.sol:63]: We may want to make this base `abstract` and have the implementing tests initialize the proxy.

That way the `v1` tests can use OZv4, and the `v2` tests can use OZv5

File: src/SuperstateToken.sol (hunk 4 above)
Comment (jonwalch) [reply src/SuperstateToken.sol:14]: will be for my next PR, this wasnt compiling before. I can remove this PR if youd like

File: test/SuperstateTokenTestBase.t.sol (hunk 6 above)
Comment (jonwalch) [reply test/SuperstateTokenTestBase.t.sol:61]: Cool if I TODO this for later?

Comment (jonwalch) [reply test/SuperstateTokenTestBase.t.sol:63]: Added the TODO

Comment (cridmann) [reply test/SuperstateTokenTestBase.t.sol:63]: 👍 should do soon though, as we're going thru a lot of trouble to maintain v1 so it should be accurate

**Review by cridmann — APPROVED** [review]

**Pull request merged**